contracts/deploy_advanced.sh

```
@@ -0,0 +1,358 @@
+    fi
+}
+
+# Function to rollback to previous deployment
+rollback() {
+    local network=$1
+    local contract_name=$2
+    
+    log_message "WARN" "Initiating rollback for $contract_name on $network"
+    
+    # Find latest backup
+    local latest_backup=$(ls -1t "$BACKUP_DIR"/${contract_name}_${network}_*.backup 2>/dev/null | head -n 1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)